File: rust/samples/src/form_card_sample.rs
use fluent_cards::*;

pub fn create_contact_form() -> Card {
    AdaptiveCardBuilder::new()
        .with_version("1.5")
        .add_text_block(|tb| {
            tb.with_text("Contact Us")
                .with_size(TextSize::Large)
                .with_weight(TextWeight::Bolder);
        })
        .add_input_text(|i| {
            i.with_id("name")
                .with_label("Name")
                .with_placeholder("Enter your name")
                .with_is_required(true)
                .with_error_message("Name is required");
        })
        .add_input_text(|i| {
            i.with_id("email")
                .with_label("Email")
                .with_placeholder("Enter your email")
                .with_style(TextInputStyle::Email)
                .with_is_required(true);
        })
        .add_input_text(|i| {
            i.with_id("message")
                .with_label("Message")
                .with_placeholder("How can we help?")
                .with_is_multiline(true)
                .with_max_length(500);
        })
        .add_action(|a| {
            a.submit(Some("Send Message"))
                .with_style(ActionStyle::Positive);
        })
        .build()
}

pub fn create_survey_form() -> Card {
    AdaptiveCardBuilder::new()
        .with_version("1.5")
        .add_text_block(|tb| {
            tb.with_text("Customer Satisfaction Survey")
                .with_size(TextSize::Large)
                .with_weight(TextWeight::Bolder);
        })
        .add_input_choice_set(|i| {
            i.with_id("satisfaction")
                .with_label("How satisfied are you?")
                .add_choice("Very Satisfied", "5")
                .add_choice("Satisfied", "4")
                .add_choice("Neutral", "3")
                .add_choice("Dissatisfied", "2")
                .add_choice("Very Dissatisfied", "1")
                .with_is_required(true);
        })
        .add_input_text(|i| {
            i.with_id("feedback")
                .with_label("Additional Feedback")
                .with_placeholder("Tell us more...")
                .with_is_multiline(true);
        })
        .add_action(|a| {
            a.submit(Some("Submit Survey"))
                .with_style(ActionStyle::Positive);
        })
        .build()
}

pub fn create_registration_form() -> Card {
    AdaptiveCardBuilder::new()
        .with_version("1.5")
        .add_text_block(|tb| {
            tb.with_text("Event Registration")
                .with_size(TextSize::Large)
                .with_weight(TextWeight::Bolder);
        })
        .add_input_text(|i| {
            i.with_id("fullName")
                .with_label("Full Name")
                .with_is_required(true);
        })
        .add_input_text(|i| {
            i.with_id("email")
                .with_label("Email Address")
                .with_style(TextInputStyle::Email)
                .with_is_required(true);
        })
        .add_input_date(|i| {
            i.with_id("eventDate").with_label("Event Date");
        })
        .add_input_toggle(|i| {
            i.with_id("newsletter")
                .with_title("Subscribe to newsletter")
                .with_value("true");
        })
        .add_action(|a| {
            a.submit(Some("Register"))
                .with_style(ActionStyle::Positive);
        })
        .build()
}

File: swift/Samples/form_card_sample.swift
import FluentCards

func createContactForm() -> Card {
    return AdaptiveCardBuilder()
        .withVersion("1.5")
        .addTextBlock { tb in
            tb.withText("Contact Us")
                .withSize(.large)
                .withWeight(.bolder)
        }
        .addInputText { i in
            i.withID("name")
                .withLabel("Name")
                .withPlaceholder("Enter your name")
                .withIsRequired(true)
                .withErrorMessage("Name is required")
        }
        .addInputText { i in
            i.withID("email")
                .withLabel("Email")
                .withPlaceholder("Enter your email")
                .withStyle(.email)
                .withIsRequired(true)
        }
        .addInputText { i in
            i.withID("message")
                .withLabel("Message")
                .withPlaceholder("How can we help?")
                .withIsMultiline(true)
                .withMaxLength(500)
        }
        .addAction { a in
            a.submit("Send Message").withStyle(.positive)
        }
        .build()
}

func createSurveyForm() -> Card {
    return AdaptiveCardBuilder()
        .withVersion("1.5")
        .addTextBlock { tb in
            tb.withText("Customer Satisfaction Survey")
                .withSize(.large)
                .withWeight(.bolder)
        }
        .addInputChoiceSet { i in
            i.withID("satisfaction")
                .withLabel("How satisfied are you?")
                .addChoice("Very Satisfied", "5")
                .addChoice("Satisfied", "4")
                .addChoice("Neutral", "3")
                .addChoice("Dissatisfied", "2")
                .addChoice("Very Dissatisfied", "1")
                .withIsRequired(true)
        }
        .addInputText { i in
            i.withID("feedback")
                .withLabel("Additional Feedback")
                .withPlaceholder("Tell us more...")
                .withIsMultiline(true)
        }
        .addAction { a in
            a.submit("Submit Survey").withStyle(.positive)
        }
        .build()
}

func createRegistrationForm() -> Card {
    return AdaptiveCardBuilder()
        .withVersion("1.5")
        .addTextBlock { tb in
            tb.withText("Event Registration")
                .withSize(.large)
                .withWeight(.bolder)
        }
        .addInputText { i in
            i.withID("fullName")
                .withLabel("Full Name")
                .withIsRequired(true)
        }
        .addInputText { i in
            i.withID("email")
                .withLabel("Email Address")
                .withStyle(.email)
                .withIsRequired(true)
        }
        .addInputDate { i in
            i.withID("eventDate")
                .withLabel("Event Date")
        }
        .addInputToggle { i in
            i.withID("newsletter")
                .withTitle("Subscribe to newsletter")
                .withValue("true")
        }
        .addAction { a in
            a.submit("Register").withStyle(.positive)
        }
        .build()
}
